v0/destinations/kafka/transform.js

```diff
@@ -1,36 +1,74 @@
+/* eslint-disable no-restricted-syntax */
+const cloneDeep = require("lodash/cloneDeep");
 const { getIntegrationsObj } = require("../../util");
 
 function batch(destEvents) {
   const respList = [];
-  const batchedRequest = [];
-  const metadata = [];
-  destEvents.forEach(event => {
-    metadata.push(event.metadata);
-    batchedRequest.push(event.message);
-  });
-  respList.push({
-    batchedRequest,
-    metadata,
-    destination: destEvents[0].destination
-  });
+
+  // Grouping the events by topic
+  const groupedEvents = destEvents.reduce((acc, event) => {
+    const { topic } = event.message;
+    if (acc[topic]) {
+      acc[topic].push(event);
+    } else {
+      acc[topic] = [event];
+    }
+    return acc;
+  }, {});
+
+  // Creating a batched request for each topic
+  // we are grouping the events based on topics
+  // and creating a batched request for each topic
+  // example: input events = [{event1,topic1},{event2,topic1},{event3,topic2}]
+  // out from transformer:  {batchedRequest:[{event1},{event2}]}, {batchedRequest:[{event3}]} (2 multilexed responses)
+  for (const [events] of Object.entries(groupedEvents)) {
```

Comment:
Object.values is sufficient right as we are not using any keys

Reply:
Addressed